internal/bgp/frr/config.go

```diff
@@ -158,26 +162,55 @@ var reloadConfig = func() error {
 // generateAndReloadConfigFile takes a 'struct frrConfig' and, using a template,
 // generates and writes a valid FRR configuration file. If this completes
 // successfully it will also force FRR to reload that configuration file.
-func generateAndReloadConfigFile(config *frrConfig) error {
+func generateAndReloadConfigFile(config *frrConfig, l log.Logger) {
 	filename, found := os.LookupEnv("FRR_CONFIG_FILE")
 	if found {
 		configFileName = filename
 	}
 
 	configString, err := templateConfig(config)
 	if err != nil {
-		return err
+		level.Error(l).Log("op", "reload", "error", err, "cause", "template", "msg", "failed to reload config", "config", config)
+		return
 	}
 
 	err = writeConfig(configString, configFileName)
 	if err != nil {
-		return err
+		level.Error(l).Log("op", "reload", "error", err, "cause", "writeConfig", "msg", "failed to reload config", "config", config)
+		return
 	}
 
 	err = reloadConfig()
 	if err != nil {
-		return err
+		level.Error(l).Log("op", "reload", "error", err, "cause", "reload", "msg", "failed to reload config", "config", config)
+		return
```

Comment:
all the above has "failed to reload config"  probably better off with different error message to differentiate the different reason

Reply:
Not sure I undestand the comment.
We'd have the err field stating what's failing.

Reply:
instead of fishing for single word difference between the different errors why not use 
```
for the 1st : "failed to setting the configuration template %s", config
for the 2nd: "failed to write config file %s",configFileName
for the 3d: "failed to reload FRR config err %s", err
```

Reply:
The log is structured, we have a cause and an err field to do that. I think we can even remove the msg because it won't add any new value.